gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java

```diff
@@ -190,18 +191,160 @@ public boolean isJobLockEnabled() {
   /**
    * Get the job state of the most recent run of the job.
    */
-  private JobState getPreviousJobState(String jobName)
-      throws IOException {
-    if (this.jobStateStore.exists(jobName, "current" + AbstractJobLauncher.JOB_STATE_STORE_TABLE_SUFFIX)) {
-      // Read the job state of the most recent run of the job
-      List<JobState> previousJobStateList =
-          this.jobStateStore.getAll(jobName, "current" + AbstractJobLauncher.JOB_STATE_STORE_TABLE_SUFFIX);
-      if (!previousJobStateList.isEmpty()) {
-        // There should be a single job state on the list if the list is not empty
-        return previousJobStateList.get(0);
+  private Map<String, JobState> getPreviousDatasetStatesByUrns(String jobName) throws IOException {
```

Comment:
What I will do is to create an empty subclass called `DatasetState` that `extends JobState`. This new class is currently identical to `JobState` without any new stuffs. However, it has a more meaningful and less confusing name.